xchange-ascendex/src/main/java/org/knowm/xchange/ascendex/AscendexAdapters.java

```diff
@@ -0,0 +1,232 @@
+package org.knowm.xchange.ascendex;
+
+import org.knowm.xchange.ascendex.dto.account.AscendexCashAccountBalanceDto;
+import org.knowm.xchange.ascendex.dto.marketdata.AscendexAssetDto;
+import org.knowm.xchange.ascendex.dto.marketdata.AscendexMarketTradesDto;
+import org.knowm.xchange.ascendex.dto.marketdata.AscendexOrderbookDto;
+import org.knowm.xchange.ascendex.dto.marketdata.AscendexProductDto;
+import org.knowm.xchange.ascendex.dto.trade.AscendexFlags;
+import org.knowm.xchange.ascendex.dto.trade.AscendexOpenOrdersResponse;
+import org.knowm.xchange.ascendex.dto.trade.AscendexPlaceOrderRequestPayload;
+import org.knowm.xchange.currency.Currency;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.account.AccountInfo;
+import org.knowm.xchange.dto.account.Balance;
+import org.knowm.xchange.dto.account.Wallet;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.marketdata.Trade;
+import org.knowm.xchange.dto.marketdata.Trades;
+import org.knowm.xchange.dto.meta.CurrencyMetaData;
+import org.knowm.xchange.dto.meta.CurrencyPairMetaData;
+import org.knowm.xchange.dto.meta.ExchangeMetaData;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.dto.trade.OpenOrders;
+import org.knowm.xchange.dto.trade.UserTrade;
+import org.knowm.xchange.dto.trade.UserTrades;
+import org.knowm.xchange.utils.jackson.CurrencyPairDeserializer;
+
+import java.time.Instant;
+import java.util.*;
+
+public class AscendexAdapters {
+
+  public static OrderBook adaptOrderBook(AscendexOrderbookDto ascendexOrderbookDto) {
+    List<LimitOrder> asks = new ArrayList<>(100);
+    List<LimitOrder> bids = new ArrayList<>(100);
```

Comment:
this isnt what i meant, new ArrayList<>(ascendexOrderbookDto.getData().getAsks().size()) for all of these, size it exactly.

Reply:
Done